python/tests/test_base_install/test_filters/semantics/test_node_property_filter_semantics.py

```diff
@@ -62,134 +14,151 @@ def init_graph_for_secondary_indexes(graph):
+        expected_ids = sorted(["N1", "N2", "N3", "N4", "N5", "N6", "N7", "N8"])
+        assert result_ids == expected_ids
 
-    filter_expr = filter.Property("p1").temporal().any() == 1
-
-    result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
-    expected_ids = sorted(["N1", "N2", "N3", "N4", "N5", "N6", "N7", "N8"])
-    assert result_ids == expected_ids
+    return check
 
 
+@with_disk_variants(init_nodes_graph, variants=["graph", "persistent_disk_graph"])
 def test_temporal_any_semantics_for_secondary_indexes():
-    graph = Graph()
-    graph = init_graph(graph)
-    graph = init_graph_for_secondary_indexes(graph)
+    def check(base_graph):
+        # Create a new graph using the same type as base_graph (Graph or PersistentGraph)
+        graph = type(base_graph)()
+        graph = init_nodes_graph(graph)
+        graph = init_graph_for_secondary_indexes(graph)
+
+        filter_expr = filter.Property("p1").temporal().any() == 1
 
-    filter_expr = filter.Property("p1").temporal().any() == 1
+        result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
+        expected_ids = sorted(["N1", "N16", "N17", "N2", "N3", "N4", "N5", "N6", "N7", "N8"])
+        assert result_ids == expected_ids
 
-    result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
-    expected_ids = sorted(["N1", "N16", "N17", "N2", "N3", "N4", "N5", "N6", "N7", "N8"])
-    assert result_ids == expected_ids
+    return check
 
 
+@with_disk_variants(init_nodes_graph, variants=["event_disk_graph", "persistent_disk_graph"])
+def test_temporal_any_semantics_for_secondary_indexes_dsg():
+    def check(graph):
+        with pytest.raises(Exception, match="Immutable graph is .. immutable!"):
+            graph = init_graph_for_secondary_indexes(graph)
+            filter_expr = filter.Property("p1").temporal().any() == 1
+            result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
+
+    return check
+
+
+@with_disk_variants(init_nodes_graph)
 def test_temporal_latest_semantics():
-    graph = Graph()
-    graph = init_graph(graph)
+    def check(graph):
+        filter_expr = filter.Property("p1").temporal().latest() == 1
+        result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
+        expected_ids = sorted(["N1", "N3", "N4", "N6", "N7"])
+        assert result_ids == expected_ids
 
-    filter_expr = filter.Property("p1").temporal().latest() == 1
-    result_ids = sorted(graph.filter_nodes(filter_expr).nodes.id)
-    expected_ids = sorted(["N1", "N3", "N4", "N6", "N7"])
-    assert result_ids == expected_ids
+    return check
 
 
+@with_disk_variants(init_nodes_graph, variants=["graph", "persistent_disk_graph"])
 def test_temporal_latest_semantics_for_secondary_indexes():
-    graph = Graph()
-    graph = init_graph(graph)
-    graph = init_graph_for_secondary_indexes(graph)
+    def check(base_graph):
+        # Create a new graph using the same type as base_graph (Graph or PersistentGraph)
```

Comment:
This is not actually testing the disk graph at all, should be done via the init func